### ISSUE TITLE
Gradient Boosting and Random Trees

### ISSUE THREAD

**Issue by FuriouslyCurious** [stepelu/lua-sci#5]
Hi Stephen,

Thank you for putting together a fantastic library!  I am a new user of Lua + Torch Machine Learning, and found SciLua while searching for good libraries.

I have a request: would it be possible to add Gradient Boosting Machine (GBM) and Random Forest or ExtremeRandom Forest algorithms to sci-lua?  It will make sci-lua much more helpful for machine learning.  

If these algorithms are already implemented in another Lua library, please let me know.

Algo descriptions: http://scikit-learn.org/stable/modules/classes.html#module-sklearn.ensemble

Thank you!

**Comment by stepelu** [comment]
I don't have immediate plans to work on this, even though I am familiar with the first 2 algorithms mentioned. 

Contributions on this are more than welcome, but ideally they would go in a separate library. Scilua is intended as a building block for scientific applications, machine/deep learning ones included.
